pyccel/parser/semantic.py

```diff
@@ -2370,27 +2380,27 @@ def _visit_FunctionDef(self, expr, **settings):
                     errors.report(msg, symbol=expr.arguments, severity='fatal')
 
         interfaces = []
-        if len(header) == 0:
-            # check if a header is imported from a header file
+        if len(headers) == 0:
+            # check if a headers is imported from a headers file
             # TODO improve in the case of multiple headers ( interface )
             func       = self.get_function(name)
             if func and func.is_header:
                 interfaces = [func]
 
-        if expr.arguments and not header and not interfaces:
+        if expr.arguments and not headers and not interfaces:
 
             # TODO ERROR wrong position
 
             errors.report(FUNCTION_TYPE_EXPECTED, symbol=name,
                    bounding_box=(self._current_fst_node.lineno, self._current_fst_node.col_offset),
                    severity='error', blocker=self.blocking)
 
-        # we construct a FunctionDef from its header
-        for hd in header:
+        # we construct a FunctionDef from its headers
+        for hd in headers:
             interfaces += hd.create_definition(templates)
```

Comment:
The comment should probably be updated: `# We construct a FunctionDef from each function header`?